### DISCUSSION TITLE
Google indexing of dynamic routes

### DISCUSSION THREAD

**Discussion by mitas1** [vercel/next.js#12438] [Help] [answered]
Hello, I'm using dynamic routes: https://nextjs.org/docs/routing/dynamic-routes. I've created the `/articles/[slug].js` file and everything works like a charm except unfortunately Google thinks that e.g. `https://<my-domain>/articles/[slug]` is a regular URL and it is showed within the google search results and links to 404. Do you have any experience with this? What is the best approach to avoid this?

**Comment by landsman** [comment, marked as answer]
1. Do you have generated sitemap for these dynamic routes? It would help.
2. Do you have correct http status code in response for these pages?

**Reply by mitas1** [reply]
Thx, I will give it a try...
